File: remind_me_fe/lib/models/dto_requests/update_todo_request.dart
class UpdateToDoRequest {
  int id;
  String title;
  String description;
  DateTime startDate;
  DateTime? endDate;
  bool isFinished;
  int difficulty;
  int ownerId;

  UpdateToDoRequest({
    required this.id,
    required this.title,
    required this.description,
    required this.startDate,
    this.endDate,
    required this.isFinished,
    required this.difficulty,
    required this.ownerId,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'isFinished': isFinished,
      'difficulty': difficulty,
      'ownerId': ownerId,
    };
  }
}

File: remind_me_be/RemindMe.ToDo/RemindMe.ToDo.Application/Requests/ToDos/UpdateToDoRequest.cs
﻿using MediatR;
using RemindMe.Domain.Entities;

namespace RemindMe.Application.Requests.Todos
{
    public class UpdateTodoRequest : IRequest<Item>
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string CreationDate { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public bool IsFinished { get; set; }
        public int Difficulty { get; set; }
        public int OwnerId { get; set; }
    }
}
